src/deadline/cinema4d_submitter/font_utils.py

```diff
@@ -0,0 +1,260 @@
+    for font_dir in font_dirs:
+        try:
+            for root, _, files in os.walk(font_dir):
+                for file in files:
+                    # Check if the font name is in the file name (case-insensitive)
+                    if font_name.lower() in file.lower():
+                        file_path = os.path.join(root, file)
+                        # Check if it's a font file by extension and exists
+                        if is_font_file(file_path) and os.path.isfile(file_path):
+                            return file_path
+        except (PermissionError, OSError) as e:
+            print(f"[ERROR] Cannot access font directory {font_dir}: {e}")
+            continue
+
+    return None
+
+
+def get_system_font_directories() -> List[str]:
+    """
+    Get the font directories based on the operating system.
+    Includes Adobe font directories and user-installed fonts.
+
+    Returns:
+        List[str]: List of font directory paths
+    """
+    system = platform.system()
+    font_dirs = []
+
+    if system == "Windows":
+        # Windows font directories
+        windir = os.environ.get("WINDIR", r"C:\Windows")
+        localappdata = os.environ.get("LOCALAPPDATA")
+        appdata = os.environ.get("APPDATA")
+
+        # System fonts
+        font_dirs = [os.path.join(windir, "Fonts")]
+
+        # User-installed fonts
+        if localappdata:
+            user_fonts_dir = os.path.join(localappdata, "Microsoft", "Windows", "Fonts")
+            font_dirs.append(user_fonts_dir)
+
+        # Adobe fonts
+        if appdata:
+            adobe_livetype = os.path.join(appdata, "Adobe", "CoreSync", "plugins", "livetype")
+            adobe_user_fonts = os.path.join(appdata, "Adobe", "User Owned Fonts")
+            font_dirs.extend([adobe_livetype, adobe_user_fonts])
+
+    elif system == "Darwin":  # macOS
+        # macOS font directories
+        font_dirs = [
+            "/Library/Fonts",
+            "/System/Library/Fonts",
+            os.path.expanduser("~/Library/Fonts"),
+        ]
+
+        # Adobe fonts on macOS
+        adobe_livetype = os.path.expanduser(
+            "~/Library/Application Support/Adobe/CoreSync/plugins/livetype"
+        )
+        adobe_user_fonts = os.path.expanduser(
+            "~/Library/Application Support/Adobe/User Owned Fonts"
+        )
+        font_dirs.extend([adobe_livetype, adobe_user_fonts])
+
+    else:
+        print(f"[ERROR] Unknown operating system: {system}")
+
+    # Filter out directories that don't exist
+    return [d for d in font_dirs if d and os.path.isdir(d)]
+
+
+def is_font_file(file_path: str) -> bool:
+    """
+    Check if a file is a font file based on its extension.
+
+    Args:
+        file_path (str): Path to the file
+
+    Returns:
+        bool: True if the file is a font file, False otherwise
+    """
+    if not file_path:
+        return False
+
+    # Adobe fonts do not have an extension.
+    font_extensions = [".otf", ".ttf", ".fon", ""]
```

Comment:
Does this treat all extension-less files as fonts? Seems like it would break something or attach the wrong files